src/patchers/NodePatcher.js

```diff
@@ -1168,4 +1168,12 @@ export default class NodePatcher {
       receiver = receiver.parent;
     }
   }
+
+  /**
+   * Gets the first "interesting token" in the indexed range (default range is `this` + parent)
+   */
+  getFirstSemanticToken(from: number=this.contentStart, to: number=this.parent.contentEnd): ?SourceToken {
```

Comment:
Yep, everything has a parent except `ProgramPatcher`, which is rarely used directly. There should be a way to do an unbounded search, at the very least by using methods on the token list directly ( https://github.com/decaffeinate/coffee-lex/blob/master/src/SourceTokenList.ts ). But yeah, the token stuff is a little clunky sometimes.